### PR TITLE
Update Diagnostic rules

### LINKED ISSUE #676: Rename "WME" in Diagnostic rules
Our current diagnostic messages all start with "WMExxxx", and changing "WME" to "CsWinRT" would make error messages clearer. The messages we report to users are DiagnosticDescriptors that get attached to Diagnostics and raised by the scanner. In WinRTRules.cs, of SourceGenerator, we need to rename the "WME" prefix of the diagnostic identifier.

### PR DESCRIPTION
**Pull request by j0shuams**
This PR introduces two new files -- AnalzyerReleases.Unshipped.md and AnalyzerReleases.Shipped.md 
These are made from our diagnostic descriptors, and serve as a documentation page for our rules. 
I am also changing the "WME" identifier of the rules to be "CsWinRT", which required a change in our diagnostic unit testing app. 

I have updated our `.gitignore` as well, as there is a new directory that we don't want to track for our winui packaging sample. 

closes #676 , addresses part of #677

### CODE REVIEW

File: src/Authoring/WinRT.SourceGenerator/WinRTRules.cs [hunk 1]
```diff
@@ -21,121 +21,130 @@ private static DiagnosticDescriptor MakeRule(string id, string title, string mes
                 helpLinkUri: "https://docs.microsoft.com/en-us/previous-versions/hh977010(v=vs.110)");
         }
 
+        public static DiagnosticDescriptor ImplementerUseSameVariableName = MakeRule(
+            "CsWinRT1113", 
+            "Class implementaiton method should use same paramter as interface method",
```
Comment (manodasanW) [inline src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:26]: typo - implementation.

Comment (manodasanW) [inline src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:26]: parameter name

File: src/Authoring/WinRT.SourceGenerator/WinRTRules.cs [hunk 2]
```diff
@@ -21,121 +21,130 @@ private static DiagnosticDescriptor MakeRule(string id, string title, string mes
...
+            "Class '{0}' implements method '{1}' with parameter '{2}', but '{1}' was declared in interface '{3}' with parameter '{4}'." 
+            + "Parameter names must match exactly in the Windows Runtime.");
+        
         public static DiagnosticDescriptor PrivateGetterRule = MakeRule(
-            "WME", 
+            "CsWinRT",
```
Comment (manodasanW) [inline src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:31]: Is there no # for this.

Comment (manodasanW) [reply src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:31]: I see a couple actually without a #.  Do we want them to all have # or no?

File: src/Authoring/WinRT.SourceGenerator/WinRTRules.cs [hunk 3]
```diff
@@ -21,121 +21,130 @@ private static DiagnosticDescriptor MakeRule(string id, string title, string mes
...
             "Class (or interface) is generic",
             "Type {0} is generic. Windows Runtime types cannot be generic.");
 
 
         public static DiagnosticDescriptor UnsealedClassRule = MakeRule(
-            "WME",
+            "CsWinRT",
             "Class is unsealed",
-            "Exporting unsealed types is not supported. Please mark type {0} as sealed.");
+            "Exporting unsealed types is not supported in Windows Runtime. Please mark type {0} as sealed.");
```
Comment (manodasanW) [inline src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:61]: I would remove the in Windows Runtime as technically it is now supported in WinRT, but we still have that limitation.

File: src/Authoring/WinRT.SourceGenerator/WinRTRules.cs [hunk 4]
```diff
@@ -21,121 +21,130 @@ private static DiagnosticDescriptor MakeRule(string id, string title, string mes
...
+            "Classes cannot have multiple constructors of the same arity in the Windows Runtime, class {0} has multiple {1}-arity constructors");
 
         public static DiagnosticDescriptor ParameterNamedValueRule = MakeRule(
-            "WME1092",
+            "CsWinRT1092",
             "Parameter Named Value Rule",
-            "The method {0} has a parameter named {1} which is the same as the default return value name. "
-            + "Consider using another name for the parameter or use the System.Runtime.InteropServices.WindowsRuntime.ReturnValueNameAttribute "
-            + "to explicitly specify the name of the return value.");
+            "The method {0} is used in the Windows Runtime and has a parameter named {1}." +
```
Comment (manodasanW) [inline src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:88]: I am not sure whether in the Windows Runtime makes sense here.

File: src/Authoring/WinRT.SourceGenerator/WinRTRules.cs (hunk 2 above)
Comment (j0shuams) [reply src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:31]: Yeah when we make the docs page I'll have a table of all of them. Was going to assign the number then, can do it now though. I can come up with numbers that aren't the ones we have already -- there's even a couple like 1060(a) that might just be clearer under a new number.

File: src/Authoring/WinRT.SourceGenerator/WinRTRules.cs (hunk 3 above)
Comment (j0shuams) [reply src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:61]: Do we want to remove the diagnostic for this then?

Comment (manodasanW) [reply src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:61]: CsWinRT still has the limitation that requires it to be sealed.

Comment (j0shuams) [reply src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:61]: Per discussion, will change to CsWinRT.

File: src/Authoring/WinRT.SourceGenerator/WinRTRules.cs [hunk 5]
```diff
@@ -18,124 +18,125 @@ private static DiagnosticDescriptor MakeRule(string id, string title, string mes
                 category: "Usage",
                 defaultSeverity: DiagnosticSeverity.Error,
                 isEnabledByDefault: true,
-                helpLinkUri: "https://docs.microsoft.com/en-us/previous-versions/hh977010(v=vs.110)");
+                helpLinkUri: "https://github.com/microsoft/CsWinRT/tree/master/src/Authoring/WinRT.SourceGenerator/AnalyzerReleases.Unshipped.md");
```
Comment (j0shuams) [inline src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:21]: We may want this to be AnalyzerReleases.Shipped.md, but I am not sure when we consider a version of our SourceGenerator analyzer "released"

File: src/Authoring/WinRT.SourceGenerator/WinRTRules.cs [hunk 6]
```diff
@@ -18,124 +18,125 @@ private static DiagnosticDescriptor MakeRule(string id, string title, string mes
...
+            "Classes cannot have multiple constructors of the same arity in the Windows Runtime, class {0} has multiple {1}-arity constructors");
 
         public static DiagnosticDescriptor ParameterNamedValueRule = MakeRule(
-            "WME1092",
+            "CsWinRT1010",
             "Parameter Named Value Rule",
-            "The method {0} has a parameter named {1} which is the same as the default return value name. "
-            + "Consider using another name for the parameter or use the System.Runtime.InteropServices.WindowsRuntime.ReturnValueNameAttribute "
-            + "to explicitly specify the name of the return value.");
+            "The method {0} is used in the Windows Runtime and has a parameter named {1}." +
```
Comment (manodasanW) [inline src/Authoring/WinRT.SourceGenerator/WinRTRules.cs:81]: Can we just merge both like The parameter name {1} in method {0} is the same as the return value used ...

**Review by manodasanW — APPROVED** [review]

File: .gitignore [hunk 7]
```diff
@@ -20,4 +20,5 @@ global.json
 nuget/Microsoft.Windows.CsWinRT.Prerelease.targets
 *.binlog
 vs_buildtools.exe
-.buildtools
\ No newline at end of file
+.buildtools
+**(Package)*
```
[the remark this answers is not in the document]
Comment (j0shuams) [reply .gitignore:24]: Hm okay I'll remove this; I put it in because of the pain of restoring many files when you accidentally add it to your commit.

**Pull request merged**